### PR TITLE
Adding metadata file notes to Micro-Manager

### PR DESCRIPTION
**Pull request by hflynn**
Explains new Micro-Manager metadata.txt file handling on MM file format page and in the dataset structure table (will need an autogen docs build run to generate table changes in the docs I think).

Sample links still be added.

### CODE REVIEW

File: components/autogen/src/format-pages.txt [hunk 1]
```diff
@@ -437,6 +437,12 @@ presenceRating = Fair
 utilityRating = Good
 reader = MicromanagerReader
 mif = true
+notes = - Bio-Formats will recognize a *metadata.txt file as a Micro-Manager format if \n
+pointed at it and will load the fileset including the companion TIFF files.\n
+- If pointed at a companion .ome.tif file, Bio-Formats will recognize an \n
+OME-TIFF format instead. This means it may load the fileset if there are \n
+multiple .ome.tif but it will not include *metadata.txt in this fileset and \n
+therefore the extended Micro-Manager metadata will be skipped.
```
Comment (melissalinkert) [inline components/autogen/src/format-pages.txt]: Needs quoting/file markup for ```*metadata.txt```?

File: components/autogen/src/format-pages.txt [hunk 2]
```diff
@@ -437,6 +437,12 @@ presenceRating = Fair
 utilityRating = Good
 reader = MicromanagerReader
 mif = true
+notes = - Bio-Formats will recognize a *metadata.txt file as a Micro-Manager format if \n
```
Comment (melissalinkert) [inline components/autogen/src/format-pages.txt]: ```as a Micro-Manager format``` reads a little weird to me - maybe ```as a Micro-Manager fileset``` or ```as being in the Micro-Manager format```?

**Comment by melissalinkert** [comment]
Looks good to me apart from the couple of minor comments above.

**Comment by hflynn** [comment]
Hopefully that reads better.
@sbesson is there a plan for the samples yet? I can add the link to this PR too if you give me a location -/images/MM/ ?

**Comment by sbesson** [comment]
I would assume the canonical link would be http://downloads.openmicroscopy.org/images/Micro-Manager/. Opened to alternate suggestions/spellings.

**Comment by hflynn** [comment]
That would work for me

**Comment by hflynn** [comment]
Links added.

File: components/autogen/src/format-pages.txt [hunk 3]
```diff
@@ -437,6 +438,13 @@ presenceRating = Fair
 utilityRating = Good
 reader = MicromanagerReader
 mif = true
+notes = - Bio-Formats will recognize a :file:`*metadata.txt` file as part of a \n
+Micro-Manager fileset if pointed at it and will load the fileset including \n
```
Comment (sbesson) [inline components/autogen/src/format-pages.txt]: These lines need to be indented (2 spaces) else the generated rst fails to build with

```
     [exec] Warning, treated as error:
     [exec] /opt/ome/bioformats/docs/sphinx/formats/micro-manager.txt:54: WARNING: Bullet list ends without a blank line; unexpected unindent.
```

**Comment by sbesson** [comment]
Argg 505dfe1 fixes the documentation build but fails the reader build with

```
[ERROR] /home/travis/build/openmicroscopy/bioformats/components/formats-bsd/src/loci/formats/in/MicromanagerReader.java:[104,49] illegal escape character
```

What about going with `A file ending with 'metadata.txt' plus one or more .tif files`?

**Comment by hflynn** [comment]
Done now @sbesson

**Comment by sbesson** [comment]
Looks good on staging and the autogeneration should be fixed. Merging.

**Comment by hflynn** [comment]
--rebased-to #2241

**Pull request merged**